File: ims-frontend/src/app/services/incident.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const API_URL = 'http://localhost:8082/v1/incidents';

export interface IncidentRequest {
  title: string;
  description: string;
  priority: string;
}

export interface Incident extends IncidentRequest {
  id: number;
  status: string;
  createdAt: string;
  createdBy?: string;
}

@Injectable({
  providedIn: 'root'
})
export class IncidentService {

  constructor(private http: HttpClient) { }

  createIncident(payload: IncidentRequest): Observable<any> {
    return this.http.post(`${API_URL}`, payload);
  }

  getAllIncidents(): Observable<any> {
    return this.http.get(`${API_URL}`);
  }

  updateStatus(id: number, status: string): Observable<any> {
    console.log("id:" + id);
    return this.http.put(`${API_URL}/${id}/status?status=${status}`, { responseType: 'text' });
  }

  purgeIncident(noOfDays: number = 30): Observable<any> {
    return this.http.delete(`${API_URL}/purge`, {
      params: new HttpParams().set('noOfDays', noOfDays.toString()),
      responseType: 'text'
    });
  }

  getRecentIncidents(): Observable<any> {
    return this.http.get(`${API_URL}/recentIncidents`);
  }
}

File: incident-service/src/main/java/com/ims/incident_service/repository/IncidentService.java
package com.ims.incident_service.repository;

import com.ims.incident_service.dto.IncidentDTO;
import com.ims.incident_service.model.Incident;
import com.ims.incident_service.model.IncidentStatus;

import java.util.List;

public interface IncidentService {
    Incident createIncident(IncidentDTO dto, String username);
    List<Incident> getUserIncidents(String username);
    List<Incident> getAllIncidents();
    Incident updateStatus(Long id, IncidentStatus status);
    int purgeResolvedIncidents(int noOfDays);
    List<Incident> getRecentIncidents(String username, boolean isAdmin);
}
